Bugsnag/Client/BugsnagClient.m

```diff
@@ -454,6 +458,7 @@ - (void)addObserverWithBlock:(BugsnagObserverBlock _Nonnull)observer {
 
     // additionally listen for metadata updates
     [self.metadata addObserverWithBlock:observer];
+    [self.metadata notifyObservers];
```

Comment:
I believe this would cause an extra 2 writes to disk of existing metadata as Client would already be registered and would get triggered when RN loads.

As a general pattern, triggering all the observers when another gets added is a little unexpected. How about adding the logic to BugsnagMetadata.notifyObservers and have it call the new observer only with a new state event?

Reply:
Updated to invoke the `observer` parameter only.